idaes/generic_models/properties/core/examples/tests/test_CO2_H2O_MEA_Ideal_VLE.py

```diff
@@ -0,0 +1,273 @@
+        model = ConcreteModel()
+        model.params = GenericParameterBlock(default=configuration)
+
+        assert isinstance(model.params.phase_list, Set)
+        assert len(model.params.phase_list) == 2
+        for i in model.params.phase_list:
+            assert i in ["Liq", "Vap"]
+        assert model.params.Liq.is_liquid_phase()
+        assert model.params.Vap.is_vapor_phase()
+
+        assert isinstance(model.params.component_list, Set)
+        assert len(model.params.component_list) == 3
+        for i in model.params.component_list:
+            assert i in ['H2O',
+                          'CO2',
+                          'MEA']
+            assert isinstance(model.params.get_component(i), Component)
+
+        assert isinstance(model.params._phase_component_set, Set)
+        assert len(model.params._phase_component_set) == 5
+        for i in model.params._phase_component_set:
+            assert i in [("Liq", "H2O"), ("Vap", "H2O"),
+                          ("Liq", "CO2"), ("Vap", "CO2"),
+                          ("Liq", "MEA")]
+
+        assert model.params.config.state_definition == FTPx
+        
+        assertStructuredAlmostEqual(
+            model.params.config.state_bounds,
+            { "flow_mol": (0, 92.0817, 100, pyunits.mol/pyunits.s),
+              "temperature": (350, 380, 500, pyunits.K),
+              "pressure": (5e4, 183430, 1e7, pyunits.Pa)},
+             item_callback=lambda x: value(x) * (
+                pyunits.get_units(x) or pyunits.dimensionless)._get_pint_unit()
+        )
+
+        assert model.params.config.phase_equilibrium_state == {
+            ("Vap", "Liq"): smooth_VLE}
+
+        assert isinstance(model.params.phase_equilibrium_idx, Set)
+        assert len(model.params.phase_equilibrium_idx) == 2
+        for i in model.params.phase_equilibrium_idx:
+            assert i in ["PE1","PE2"]
+
+        assert model.params.phase_equilibrium_list == {
+            "PE1": {"H2O": ("Vap", "Liq")},
+            "PE2": {"CO2": ("Vap", "Liq")}}
+
+        assert model.params.pressure_ref.value == 101325
+        assert model.params.temperature_ref.value == 298.15
+
+        assert model.params.H2O.mw.value == 18.0153E-3
+        assert model.params.H2O.pressure_crit.value == 220.64E5
+        assert model.params.H2O.temperature_crit.value == 647
+
+        assert model.params.CO2.mw.value == 44.0095E-3
+        assert model.params.CO2.pressure_crit.value == 73.825E5
+        assert model.params.CO2.temperature_crit.value == 304.23
+
+        assert_units_consistent(model)
+
+
+class TestStateBlock(object):
+    @pytest.fixture(scope="class")
+    def model(self):
+        model = ConcreteModel()
+        model.params = GenericParameterBlock(default=configuration)
+
+        model.props = model.params.build_state_block(
+                [1],
+                default={"defined_state": True})
+
+        return model
+
+    @pytest.mark.unit
+    def test_build(self, model):
```

Comment:
Examples don't really need to test most of this stuff. Just focus on an application case and check the results.